### PR TITLE
chore: lead/expert lvl + filters refactor + dropdowns refactor

### PR DESCRIPTION
**Pull request by AniaPiWo**

### CODE REVIEW

**Review by NerdbordMentor — CHANGES_REQUESTED** [review]
## Code Review Report

### Task Requirements
The task requirement was to add an option to choose "Lead/Expert" in the seniority level selection within the Work Information component.

### Review of Code Changes

#### `src/components/CreateProfile/WorkInformation/WorkInformation.tsx`
- **Requirement Fulfillment**: The code change in `WorkInformation.tsx` replaces 'Senior' with 'Lead / Expert' in the `seniority` array within the `filterLists` object. This change fulfills the task requirement by providing an option to choose "Lead/Expert" as a seniority level.
  
  ```diff
  @@ -11,7 +11,7 @@
   const filterLists = {
  -  seniority: ['Intern', 'Junior', 'Mid', 'Senior'],
  +  seniority: ['Intern', 'Junior', 'Mid', 'Lead / Expert'],
   }
  ```
  
  However, it appears that the original 'Senior' option has been removed. If the intention was to add "Lead/Expert" in addition to the existing options, this would be a bug. The corrected code should include both 'Senior' and 'Lead / Expert':

  ```javascript
  const filterLists = {
    seniority: ['Intern', 'Junior', 'Mid', 'Senior', 'Lead / Expert'],
    // ... other properties
  };
  ```

#### Other Files (`loading.tsx` in various directories)
- **Requirement Fulfillment**: The addition of `loading.tsx` files in various directories (`src/app/github-oauth`, `src/app/moderation`, `src/app/moderation/profile/[id]`, `src/app/my-profile/edit`, `src/app/profiles/[id]`) is not related to the task requirement. These files seem to be implementing a loading component, which is a separate concern from the task at hand.

### Conclusion
The task requirement of adding an option to choose "Lead/Expert" is partially met in the `WorkInformation.tsx` file. However, if the intention was to add this option without removing 'Senior', then there is a bug in the implementation. The other code changes regarding loading components are unrelated to the task and do not impact the review of the task requirement.

File: src/data/frontend/profile/mappers.ts [hunk 1]
```diff
@@ -14,3 +15,56 @@ export const mapEmploymentType = (employmentType: EmploymentType) => {
...
+
+// TODO: Wypluć dynamiczmnie
+export const mappedLocations = ['Poland', 'Europe', 'Other'].map(
+  (location) => ({
+    name: location,
+    value: location,
+  }),
+)
+
+// TODO: Szymon tu będzie działał
```
Comment (szymonkadas) [inline src/data/frontend/profile/mappers.ts:58]: noted.

**Review by szymonkadas — APPROVED** [review]
Love it <3

File: prisma/schema.prisma [hunk 2]
```diff
@@ -13,35 +13,35 @@ datasource db {
...
+  bio                      String
+  country                  Country           @relation(fields: [countryId], references: [id])
+  countryId                String
+  openForCountryRelocation Boolean           @default(false)
+  city                     City              @relation(fields: [cityId], references: [id])
+  cityId                   String
+  openForCityRelocation    Boolean           @default(false)
+  remoteOnly               Boolean
+  position                 Specialization
+  seniority                SeniorityLevel
```
Comment (mmcode9407) [inline prisma/schema.prisma:40]: You have to change it on string or string[]

Comment (szymonkadas) [reply prisma/schema.prisma:40]: I'll implement this, cuz for now multiselects aren't merged yet

Comment (mmcode9407) [reply prisma/schema.prisma:40]: Ohkee dohkee :)

File: src/components/CreateProfile/LocationPreferences/LocationPreferences.tsx [hunk 3]
```diff
@@ -1,78 +1,93 @@
-'use client'
-import React from 'react'
-import styles from './LocationPreferences.module.scss'
-import TextInput from '@/components/TextInput/TextInput'
-import CheckboxInput from '@/components/Checkbox/Checkbox'
-import SwitchInput from '@/components/Switch/Switch'
 import { useFormikContext } from 'formik'
```
Comment (mmcode9407) [inline src/components/CreateProfile/LocationPreferences/LocationPreferences.tsx:1]: `use client` missed

File: src/components/CreateProfile/LocationPreferences/LocationPreferences.tsx [hunk 4]
```diff
@@ -1,78 +1,93 @@
...
   return (
     <div className={styles.container}>
-      <div className={styles.left}>
-        <div>Location preferences</div>
-        <div className={styles.personalInfo}>
-          Share your current qualifications information. You’ll be able to
-          change it at any moment.
-        </div>
-      </div>
+      [...]
```
Comment (mmcode9407) [inline src/components/CreateProfile/LocationPreferences/LocationPreferences.tsx:25]: probably accidentally deleted div

File: src/components/CreateProfile/WorkInformation/WorkInformation.tsx [hunk 5]
```diff
@@ -59,35 +58,49 @@ const WorkInformation = () => {
...
+            name={WorkInformationFormKeys.SENIORITY}
           />
         </InputFormError>
-        <InputFormError error={touched.techStack && errors.techStack}>
+        <InputFormError
+          /* TODO: Fix it */
+          // error={
+          //   touched[WorkInformationFormKeys.TECH_STACK] &&
+          //   errors[WorkInformationFormKeys.TECH_STACK]
+          // }
```
Comment (mmcode9407) [inline src/components/CreateProfile/WorkInformation/WorkInformation.tsx:96]: I think the error is connected with the Yup schema. We have an array of strings there but in fact, we have an array of objects `DropdownOption`. We should change the techStack validation schema

**Review by mmcode9407 — COMMENTED** [review]
I think we also need to change the techStack type in ProfileModel becouse there is an array of strings

**Review by Sumick — APPROVED** [review]
Ready to merge - however let's merge previous little PRs before that one goes in to avoid conflicts

File: src/components/CreateProfile/LocationPreferences/LocationPreferences.tsx [hunk 6]
```diff
@@ -24,55 +39,79 @@ const LocationPreferences = () => {
...
-            checked={values.openToRelocationCity}
-            onChange={handleChange}
-            name="openToRelocationCity"
-            dataTestId="openToRelocationCity"
+            checked={
+              values[LocationPreferencesFormKeys.OPEN_TO_RELOCATION_CITY]
+            }
+            onChange={() => {
+              handleToggleCheckbox(
+                '1',
```
Comment (szymonkadas) [inline src/components/CreateProfile/LocationPreferences/LocationPreferences.tsx:100]: issue, this function shouldn't use string, i've changed it to boolean. ('1' is a string not boolean required in schema)

File: src/lib/discord.ts [hunk 7]
```diff
@@ -17,10 +17,10 @@ export async function sendDiscordNotificationToWebhook(params: {
 export async function sendDiscordNotificationToModeratorChannel(
   message: string,
 ) {
-  if (process.env.MODERATION_WEBHOOK) {
-    sendDiscordNotificationToWebhook({
+  if (process.env.NEXT_PUBLIC_MODERATION_WEBHOOK) {
+    await sendDiscordNotificationToWebhook({
       message,
-      webhookUrl: process.env.MODERATION_WEBHOOK,
+      webhookUrl: process.env.NEXT_PUBLIC_MODERATION_WEBHOOK,
```
Comment (szymonkadas) [inline src/lib/discord.ts:23]: please provide on pv env var

File: src/data/frontend/profile/mappers.ts [hunk 8]
```diff
@@ -1,16 +1,90 @@
...
+      return 'Backend'
+    case JobSpecialization.Fullstack:
+      return 'Fullstack'
     default:
-      return employmentType
+      return specialization
   }
 }
+
+export const mappedEmploymentType = mapEmploymentTypes(
```
Comment (szymonkadas) [inline src/data/frontend/profile/mappers.ts:52]: issue: map isn't returning dropdownElement compatible type (fixed)

**Review by szymonkadas — CHANGES_REQUESTED** [review]
There were some issues mainly due to loss of my previous multi select dropdown function (maybe it got overwrtitten in merges?).
Issues: filters display, functionality, profile creation (technologies were using bad typing) 
Things to do in future:
- technologies in filters should be fetched from db.
Things to consider: 
- filters tasks in profileList (how they should filter data)

**Comment by szymonkadas** [comment]
@Sumick it should be working now, but i insist on reviewing these changes.

**Review by wawrzonnn — APPROVED** [review]

**Review by szymonkadas — APPROVED** [review]
Should be fine.

**Pull request merged**